### DISCUSSION TITLE
Learning | Drupal Book

### DISCUSSION THREAD

**Discussion by giscus** [selwynpolit/d9book#67] [Q&A]
# Learning | Drupal Book

Drupal at your fingertips

https://selwynpolit.github.io/d9book/learn

**Comment by carlosilipo77** [comment]
Hi, 
I’ve been trying to learn the internals of Drupal. Hence, I bought the book on module development. I’m not that impressed. Once again, the book assumes your seasoned Drupal developer, just looking for a few insights. 

Whilst, I seen other books ( online) go into great lengths where the classes live and operate from.

I studied Drupal back in the old days of 7/early 8, and now I’m having a steep learning curve for Drupal 9/10. It’s not the modules, but how to develop a website for  friend’s business. 

There does not seem to be any resource, for an average developer to get up running a website with Drupal 9/10 that is; all the bits like a hero, a footer, content..etc, using whatever framework ( bootstrap, DXPR or others).  We have new layout module, paragraphs, but once again - how does that translate into a website.  Think of it like this. I used to be commi-chef ( when I was younger).  If I was to teach you how to make a beef stock;  roast the bones, brown the onions & vegetables, fry the vegetables, add the tomato paste…. If I show you - you’ll be able to make your own stock. 

Once again, there are lot of resources out there - but we need someone to sew them all together, so as to make sense. 

Sorry for the rant!
Carlo

**Reply by selwynpolit** [reply]
Hi Carlo, I added links to a few more books on this page.  Hopefully you find something that helps you tie it all together.

**Comment by Elaine-dev** [comment]
My biggest complaints about learning Drupal in general is
* Inconsistency in usage of terms
* Conflicting information, esp. with dependency injection

**Reply by selwynpolit** [reply]
Hopefully none of the information in this book is conflicting.  Please let me know if you find any.  I'm working hard to try to make it pretty clear.
